mobile/src/main/res/xml/preferences.xml

```diff
@@ -6,61 +6,81 @@
             android:key="default_openhab_demomode"
             android:summary="@string/settings_openhab_demomode_summary"
-            android:title="@string/settings_openhab_demomode" />
-        <org.openhab.habdroid.ui.widget.URLInputPreference
-            android:defaultValue="@string/empty_string"
-            android:dependency="default_openhab_demomode"
-            android:inputType="textNoSuggestions"
-            android:key="default_openhab_url"
-            android:summary="@string/settings_openhab_url_summary"
-            android:title="@string/settings_openhab_url" />
-        <org.openhab.habdroid.ui.widget.URLInputPreference
-            android:defaultValue="@string/empty_string"
-            android:dependency="default_openhab_demomode"
-            android:inputType="textNoSuggestions"
-            android:key="default_openhab_alturl"
-            android:summary="@string/settings_openhab_alturl_summary"
-            android:title="@string/settings_openhab_alturl" />
-        <EditTextPreference
-            android:defaultValue="@string/empty_string"
-            android:dependency="default_openhab_demomode"
-            android:inputType="textNoSuggestions"
-            android:key="default_openhab_username"
-            android:title="@string/settings_openhab_username" />
-        <EditTextPreference
-            android:defaultValue="@string/empty_string"
-            android:dependency="default_openhab_demomode"
-            android:inputType="textPassword"
-            android:key="default_openhab_password"
-            android:title="@string/settings_openhab_password" />
-        <Preference
-            android:clickable="true"
-            android:defaultValue="@string/settings_openhab_none"
-            android:dependency="default_openhab_demomode"
-            android:key="default_openhab_sslclientcert"
-            android:summary=""
-            android:title="@string/settings_openhab_sslclientcert" />
-        <Preference
-            android:clickable="true"
-            android:dependency="default_openhab_demomode"
-            android:key="default_openhab_sslclientcert_howto"
-            android:summary="@string/settings_openhab_sslclientcert_howto_summary"
-            android:title="@string/settings_openhab_sslclientcert_howto" />
-        <CheckBoxPreference
-            android:defaultValue="false"
-            android:key="default_openhab_sslcert"
-            android:summary="@string/settings_openhab_sslcert_summary"
-            android:title="@string/settings_openhab_sslcert" />
-        <CheckBoxPreference
-            android:defaultValue="false"
-            android:key="default_openhab_sslhost"
-            android:summary="@string/settings_openhab_sslhost_summary"
-            android:title="@string/settings_openhab_sslhost" />
+            android:title="@string/settings_openhab_demomode"/>
+        <PreferenceScreen android:title="@string/settings_openhab_connection">
```

Comment:
I find a mixture of categories and screens a bit weird. Not sure whether this + the Google guidelines count as good reason in your book ;-)